docs/source/conf.py

```diff
@@ -312,8 +320,8 @@
     ],
     "show_version_warning_banner": True,
     "switcher": {
-        "json_url": "/docs/_static/versions.json",
-        "version_match": version,
+        "json_url": "https://arrow.apache.org/docs/_static/versions.json",
```

Comment:
Do we want to keep this one as it was (that's in theory unrelated to the rest of this PR) ?

Reply:
I do not mind changing back. It is much easier to test though. For anybody not knowing why the version_switcher is not working locally - and on the PR comment also.

Reply:
Reverted the change: https://github.com/apache/arrow/pull/38241/commits/26a6561e144cfda2beec5724cc832a25dabde11c

Reply:
OK, yes, let's leave this change then. 

I think the main benefit is that it doesn't tie us to the exact base url (since it's always pointing relative to the root url), like if arrow.apache.org would change to something else. Which is quite unlikely on the short term, and even if it happens, it's also just a simple patch to the older doc built sources to replace `DOCUMENTATION_OPTIONS.theme_switcher_json_url = "...";`

Reply:
Can change the link of the "json_url" separately.

Reply:
> I think the main benefit is that it doesn't tie us to the exact base url (since it's always pointing relative to the root url), like if arrow.apache.org would change to something else. Which is quite unlikely on the short term, and even if it happens, it's also just a simple patch to the older doc built sources to replace `DOCUMENTATION_OPTIONS.theme_switcher_json_url = "...";`

Yes, I agree. We synced the work so the change for "json_url" is not added, but can be later if we see fit.

Reply:
Ah, merged a few seconds too late to _not_ include your last commit ;)